app/assets/javascripts/darkswarm/services/groups.js.coffee

```diff
@@ -1,14 +1,3 @@
-Darkswarm.factory 'Groups', (groups, Enterprises, Dereferencer) ->
+Darkswarm.factory 'Groups', (groups) ->
   new class Groups
     groups: groups
-    groups_by_id: {} 
-    constructor: ->
-      for group in @groups
-        @groups_by_id[group.id] = group
-      @dereference()
-    dereference: ->
-      for group in @groups
-        Dereferencer.dereference group.enterprises, Enterprises.enterprises_by_id 
-      for enterprise in Enterprises.enterprises
-        Dereferencer.dereference enterprise.groups, @groups_by_id
```

Comment:
:+1: good to see one more instance of this Dereferencering thing going away! We only have 4 or 5 more usages of it...

Reply:
Yes, I've come to realise that anywhere this `Dereferencer` is being used in Angular, I can be sure that:
- something horrible is happening
- it's impossible to follow what it's doing

Reply:
LOLOL
it's documented here: https://github.com/openfoodfoundation/openfoodnetwork/wiki/Angular-and-OFN#dereferencing